### ISSUE TITLE
Unable to run yarn build:api-reports on Windows

### ISSUE THREAD

**Issue by awanlin** [backstage/backstage#7198]
## Expected Behavior

Running `yarn build:api-reports` should work and generate the reports as needed on all environments including Windows

## Current Behavior

When we run `yarn  build:api-reports` on Windows we are getting the following error even though storybook has been excluded

```shell
Error: Error parsing C:\forks\backstage\packages\storybook:
The "mainEntryPointFilePath" path does not exist: C:\forks\backstage\dist-types\packages\storybook\src\index.d.ts
    at Function.prepare (C:\forks\backstage\node_modules\@microsoft\api-extractor\src\api\ExtractorConfig.ts:976:13)
    at runApiExtraction (C:\forks\backstage\scripts\api-extractor.ts:162:45)
    at main (C:\forks\backstage\scripts\api-extractor.ts:549:3)
```

## Possible Solution

We found that changing the `/` to `\\` for `SKIPPED_PACKAGES` paths here: https://github.com/backstage/backstage/blob/d0d78ebc7cf12d2d9eef5ebe50547000d0bc70d5/scripts/api-extractor.ts#L95

This got things working on Windows but that is not a permanent fix for this.

We did also try changing this line: https://github.com/backstage/backstage/blob/d0d78ebc7cf12d2d9eef5ebe50547000d0bc70d5/scripts/api-extractor.ts#L128

To this: `if (!SKIPPED_PACKAGES.includes(normalize(packageDir))) {` but this did not work as the same error came up. `normalize` is from the `path` package

## Steps to Reproduce

In a Windows environment:

1. Clone Backstage
2. Run `yarn install`, `yarn tsc`, and `yarn build`
3. Now run `yarn  build:api-reports`

Notice that you get and error similar to this:

```shell
Error: Error parsing C:\forks\backstage\packages\storybook:
The "mainEntryPointFilePath" path does not exist: C:\forks\backstage\dist-types\packages\storybook\src\index.d.ts
    at Function.prepare (C:\forks\backstage\node_modules\@microsoft\api-extractor\src\api\ExtractorConfig.ts:976:13)
    at runApiExtraction (C:\forks\backstage\scripts\api-extractor.ts:162:45)
    at main (C:\forks\backstage\scripts\api-extractor.ts:549:3)
```

## Context

I submitted a PR and one of the related builds fails due to the API Report being missing. I then tried to run the command to create it when is when I ran into this error. this is blocking me from fixing my PR.

## Your Environment

The key to this issue is being on Windows

**Comment by awanlin** [comment]
I've also tried `if (!SKIPPED_PACKAGES.includes(win32.normalize(packageDir))) {` but that results in the same error as well

**Comment by Rugvip** [comment]
Thank you for reporting!

Shipping a fix in #7202. The final tricky bit is that Node does some weird sh*t on Windows when resolving a relative path between a relative and an absolute one. You end up with a somewhat unexpected relative path that broke the comparison. Making sure the relative path resolution was between two absolute paths fixes it.

**Comment by awanlin** [comment]
Thanks @Rugvip! And also @timbonicus for the help in Discord on this. From looking at your fix this was clearly not something I was going to figure out on my own.
